src/PowerShellEditorServices/Services/DebugAdapter/Handlers/StackTraceHandler.cs

```diff
@@ -1,64 +1,132 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+#nullable enable
 
 using System;
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Management.Automation;
 using Microsoft.PowerShell.EditorServices.Services;
-using Microsoft.PowerShell.EditorServices.Services.DebugAdapter;
-using Microsoft.PowerShell.EditorServices.Utility;
 using OmniSharp.Extensions.DebugAdapter.Protocol.Models;
 using OmniSharp.Extensions.DebugAdapter.Protocol.Requests;
+using Microsoft.PowerShell.EditorServices.Services.DebugAdapter;
+using System.Linq;
+
+namespace Microsoft.PowerShell.EditorServices.Handlers;
 
-namespace Microsoft.PowerShell.EditorServices.Handlers
+internal class StackTraceHandler(DebugService debugService) : IStackTraceHandler
 {
-    internal class StackTraceHandler : IStackTraceHandler
+    /// <summary>
+    /// Because we don't know the size of the stacktrace beforehand, we will tell the client that there are more frames available, this is effectively a paging size, as the client should request this many frames after the first one.
+    /// </summary>
+    private const int INITIAL_PAGE_SIZE = 20;
+
+    public async Task<StackTraceResponse> Handle(StackTraceArguments request, CancellationToken cancellationToken)
     {
-        private readonly DebugService _debugService;
+        if (!debugService.IsDebuggerStopped)
+        {
+            throw new NotSupportedException("Stacktrace was requested while we are not stopped at a breakpoint.");
```

Comment:
Patrick may have also said this but we should be sure in the cases where we throw that we _really mean to throw_ because if not handled, it'll crash (and so often it's better to just log a warning and return if that's more graceful).

Reply:
Fair, this was supposed to be one of those "this should never happen" exceptions that if it happens its 99.99% chance its a bug and we *want* it to be reported. I was going to update the handling of that in my reporter updates but for now I can do that.

Reply:
I converted it to RPCErrorException so that it only errors out the handler. I have a HandlerErrorException in the rename PR that really simplifies this kind of stuff.